packages/docs/content/docs/tutorials/how-we-created-nes-model.mdx

```diff
@@ -0,0 +1,87 @@
+
+NES keeps offering the next edit, then the one after that, and then the one after that. You can walk through these changes across all your files, tapping `Tab` and stacking up saved keystrokes (and time!) with every acceptance.
+
+![NES Demo](./images/nes-demo-blog.gif)
+
+
+Technically this is much harder to achieve, since NES considers the entire file plus your recent edit history and predicts how your code is likely to evolve: where the next change should happen, and what that change should be. 
+
+Other tools have versions of this idea, but models have evolved a lot, and so has our understanding of how people actually write code. So we bring you our own take on Pochi’s NES: a bit different, a bit more opinionated.  This is the first post in our "Next Edit Suggestion" series, where we’ll share how we trained, evaluated, and optimized our internal Next Edit model in Pochi.
+
+## How We Train the Edit Model
+
+When we decided to build NES, one of the first pressing questions on our mind was: What kind of data actually teaches a model to make good edits?
+
+It turned out that real developer intent is surprisingly hard to capture. As anyone who’s peeked at real commits knows, developer edits are messy. Pull requests bundle unrelated changes, commit histories jump around, and the sequences of edits often skip the small, incremental steps engineers actually take when exploring or fixing code.
+
+### The Training Process
+
+To train an edit model, we format each example using special edit tokens. These tokens tell the model: 
+
+- What part of the file is editable
+- The user’s cursor position
+- What the user has edited so far
+- What the next edit should be inside that region only
+
+Unlike chat-style models that generate free-form text, NES is trained to predict the next code edit inside the editable region.
+
+Below is an example of how NES predicts the next edit:
+
+![SFT conversion image](./images/sft-conversion-image.png)
+
+In the image above, the developer makes the first edit allowing the model to capture the intent of the user. The `editable_region` markers define everything between them as the editable zone. The `user_cursor_is_here` token shows the model where the user is currently editing. 
+
+NES infers the transformation pattern (capitalization in this case) and applies it consistently as the next edit sequence. 
+
+To support this training format, we used **[CommitPackFT](https://huggingface.co/datasets/lurf21/NextEditPrediction)** and **[Zeta](https://huggingface.co/datasets/zed-industries/zeta)** as data sources. We normalized this unified dataset into the same Zeta-derived edit-markup format as described above and applied filtering to remove non-sequential edits using a small in-context model (GPT-4.1 mini).
```

Comment:
https://huggingface.co/datasets/bigcode/commitpackft

this is correct link of commitpack ft